src/content/en/fundamentals/engage-and-retain/push-notifications/web-push-protocol.md

```diff
@@ -0,0 +1,613 @@
+
+The 16 byte salt should be base64 URL safe encoded and added to the Encryption header, like so:
+
+    Encryption: salt=<URL Safe Base64 Encoded Salt>
+    
+#### Crypto-Key Header
+
+We saw that the 'Crypto-Key' header is used under the 'Application Server Keys'
+section to contain the public application server key.
+
+This header is also used to share the local public key used to encrypt
+the payload.
+
+The resulting header looks like this:
+
+    Crypto-Key: dh=<URL Safe Base64 Encoded Local Public Key String>; p256ecdsa=<URL Safe Base64
+    Encoded Public Application Server Key>
+    
+#### Content Type, Length & Encoding Headers
+
+The 'Content-Length' header is the number of bytes in the encrypted payload and 'Content-Type'
+and 'Content-Encoding' headers are fixed values, as shown below:
+
+    Content-Length: <Number of Bytes in Encrypted Payload>
+    Content-Type: 'application/octet-stream'
+    Content-Encoding: 'aesgcm'
+    
+With these headers set, we need to send the encrypted payload as the body of
+our request. Notice that the Content-Type is set to
+'application/octet-stream'. This is because the encrypted payload must be sent as a stream of
+bytes.
+
+In NodeJS we would do this like so:
+
+    const pushRequest = https.request(httpsOptions, function(pushResponse) {
+    pushRequest.write(encryptedPayload);
+    pushRequest.end();
+    
+## More Headers?
+
+We've covered the headers used for JWT / Application Server Keys (i.e. how to identify the
+application with the push service) and we've covered the headers used to send an encrypted
+payload.
+
+There are some additional headers that can be used by Push Services to alter the behavior of
+the send messages. Some of these headers are required, while others are optional.
+
+### TTL Header
+
+This is a **required header**.
+
+TTL (or time to live) should be an integer for the number of seconds you
+want your push message to live on the push service before it's delivered. If
+the TTL is expired, the message will be removed from the Push Service queue
+and it won't be delivered.
+
+    TTL: <Time to live in seconds>
+
+If you set a TTL of zero, the push service will attempt to deliver the message
+immediately **but** if the device can't be reached, your message will be
+immediately dropped from the push service queue.
+
+Technically a push service can reduce the TTL of a push message if it wants. You can tell if
+this has happened by examining the TTL header in the response from a push service.
+
+##### Topic
+
+This is an **optional header**.
+
+Topics are strings that can be used to replace any pending notifications with
+new notifications if they have matching topic names.
+
+This is useful in scenarios where the user's device might have been offline during which you
+might have sent three messages and you may want the user to receive only the latest message
+when their device is eventually turned back on.
```

Comment:
Suggested rewrite:

This is useful in scenarios where multiple messages are sent while a device is offline, and you really only want a user to see the last one when the device is turned on.